packages/protocol/thread/analysis.ts

```diff
@@ -0,0 +1,91 @@
+import { AnalysisEntry, Location, PointDescription } from "@replayio/protocol";
+import { AnalysisParams } from "../analysisManager";
+import { gAnalysisCallbacks, sendMessage } from "../socket";
+
+export interface Analysis {
+  analysisId: string;
+  addLocation: (location: Location) => Promise<void>;
+  findPoints: () => Promise<FindAnalysisPointsResult>;
+  runAnalysis: () => Promise<RunAnalysisResult>;
+  releaseAnalysis: () => Promise<void>;
+}
+
+export enum AnalysisError {
+  TooManyPointsToFind,
+  TooManyPointsToRun,
+}
+
+interface FindAnalysisPointsResult {
+  points: PointDescription[];
+  error: AnalysisError.TooManyPointsToFind | undefined;
+}
+
+interface RunAnalysisResult {
+  results: AnalysisEntry[];
+  error: AnalysisError.TooManyPointsToRun | undefined;
+}
+
+export const createAnalysis = async (params: AnalysisParams): Promise<Analysis> => {
+  // Call to the client and say hey please make an analysis and after that
+  // create an Analysis with that result
+  const { analysisId } = await sendMessage(
+    "Analysis.createAnalysis",
+    {
+      mapper: params.mapper,
+      reducer: params.reducer,
+      effectful: params.effectful,
+    },
+    params.sessionId
+  );
+  const points: PointDescription[] = [];
+  const results: AnalysisEntry[] = [];
+  gAnalysisCallbacks.set(analysisId, {
+    onEvent: receivedPoints => {
+      points.push(...receivedPoints);
+    },
+    onCreate: console.log,
+    onResults: receivedResults => {
+      results.push(...receivedResults);
+    },
+  });
+  return {
+    analysisId,
+    async findPoints() {
+      try {
+        await sendMessage("Analysis.findAnalysisPoints", { analysisId }, params.sessionId);
```

Comment:
This could be `Analysis.findAnalysisPoints()`

Reply:
I've been wondering why we sometimes use one and sometimes the other.